# Data Management and Visualization
### Week 3: Running Your First Program

For the next step in my data analysis project, I needed to code out several rows in my data set for which data is not available. For example, the variable for internet user rate for Taiwan is simply a blank cell with no data recorded; I replaced all of these non-values with a Numpy NaN. The second data management step I took was to bin the data into ten bins (except for the polity score, which was recorded in integer values from -10 to 10, resulting 21 bins). This will allow me to print a frequency distribution, or  a histogram later on.

The first step is to import the data and create a new dataframe with the binned variables.

In [5]:
'''
Code for Peer-graded Assignment: Making Data Management Decisions
Course: Data Management and Visualization
Specialization: Data Analysis and Interpretation
'''
 
import pandas as pd
 
data = pd.read_csv('C:/Users/Greg/desktop/gapminder.csv', low_memory=False)

bindata = data.copy()
 
data['internetuserate'] = pd.to_numeric(data['internetuserate'], errors='coerce')
bindata['internetuserate'] = pd.cut(data.internetuserate, 10)
 
data['incomeperperson'] = pd.to_numeric(data['incomeperperson'], errors='coerce')
bindata['incomeperperson'] = pd.cut(data.incomeperperson, 10)
 
data['employrate'] = pd.to_numeric(data['employrate'], errors='coerce')
bindata['employrate'] = pd.cut(data.employrate, 10)
 
data['femaleemployrate'] = pd.to_numeric(data['femaleemployrate'], errors='coerce')
bindata['femaleemployrate'] = pd.cut(data.femaleemployrate, 10)
 
data['polityscore'] = pd.to_numeric(data['polityscore'], errors='coerce')
bindata['polityscore'] = data['polityscore']
 
sub2 = bindata.copy()

Printing the frequency distributions

In [6]:
print ('counts for internetuserate – Percentage of people with access to internet')
c1 = sub2['internetuserate'].value_counts(sort=False)
print (c1)
print ()
 
print ('percentages for internetuserate - Percentage of people with access to internet')
p1 = sub2['internetuserate'].value_counts(sort=False, normalize=True)
print (p1)
print ()
 
print ('counts for incomeperperson - GDP per capita')
c2 = sub2['incomeperperson'].value_counts(sort=False)
print(c2)
print ()
 
print ('percentages for incomeperperson - GDP per capita')
p2 = sub2['incomeperperson'].value_counts(sort=False, normalize=True)
print (p2)
print ()
 
print ('counts for employrate – Percentage of people age 15+ who were employed in the year')
c3 = sub2['employrate'].value_counts(sort=False)
print(c3)
print ()
 
print ('percentages for employrate - Percentage of people age 15+ who were employed in the year')
p3 = sub2['employrate'].value_counts(sort=False, normalize=True)
print (p3)
print ()
 
print ('counts for femaleemployrate - Percentage of females age 15+ who were employed in the year')
c4 = sub2['femaleemployrate'].value_counts(sort=False)
print(c4)
print ()
 
print ('percentages for femaleemployrate - Percentage of females age 15+ who were employed in the year')
p4 = sub2['femaleemployrate'].value_counts(sort=False, normalize=True)
print (p4)
print ()
 
print ("counts for polityscore - The summary measure of a country's democratic and free nature. -10 is the lowest value, 10 the highest.")
c5 = sub2['polityscore'].value_counts(sort=False, bins=20)
print(c5)
print ()
 
print ("percentages for polityscore - The summary measure of a country's democratic and free nature. -10 is the lowest value, 10 the highest.")
p5 = sub2['polityscore'].value_counts(sort=False, normalize=True, bins=20)
print (p5)
print ()

counts for internetuserate – Percentage of people with access to internet
(0.115, 9.753]       46
(9.753, 19.296]      27
(19.296, 28.838]     17
(28.838, 38.381]     18
(38.381, 47.924]     24
(47.924, 57.467]     13
(57.467, 67.0097]    11
(67.0097, 76.553]    13
(76.553, 86.0953]    16
(86.0953, 95.638]     7
Name: internetuserate, dtype: int64

percentages for internetuserate - Percentage of people with access to internet
(0.115, 9.753]       0.239583
(9.753, 19.296]      0.140625
(19.296, 28.838]     0.088542
(28.838, 38.381]     0.093750
(38.381, 47.924]     0.125000
(47.924, 57.467]     0.067708
(57.467, 67.0097]    0.057292
(67.0097, 76.553]    0.067708
(76.553, 86.0953]    0.083333
(86.0953, 95.638]    0.036458
Name: internetuserate, dtype: float64

counts for incomeperperson - GDP per capita
(-1.268, 10608.142]         144
(10608.142, 21112.508]       18
(21112.508, 31616.874]       13
(31616.874, 42121.241]       11
(42121.241, 52625.607]        1
(52625.607, 63129.973]     

Of the first three variables, all of them contained missing data. In fact, for all three variables, almost 20 data points were missing out of more than 210 rows–and it was not consistently the same rows, an issue which I will address in a later week.

The variable for internet user rate is skewed left, with nearly 25% of values falling in the first decile. For these countries in the first decile, between roughly 0% and 10% of the population has access to the internet.

It is a similar story for GDP per capita, but much more heavily skewed left. More than 75% of all the data falls in the first decile! According to the data, 75% of the countries surveyed have a GDP per capita of less than 10,000 USD, adjusted for inflation. Only 1 country has a GDP per capita higher than 100,000 USD.

The employment rate variable is much more bell-shaped, approaching a normal distribution. 19% of the countries surveyed have employment rates around 55% and 20% of countries have employment rates of 60%. In total, this is 72 countries with employment rates in the high 50%’s.